solidity/contracts/bridge/Wallets.sol

```diff
@@ -537,6 +560,68 @@ library Wallets {
         self.registry.closeWallet(wallet.ecdsaWalletID);
     }
 
+    function submitMovingFundsCommitment(
+        Data storage self,
+        bytes20 walletPubKeyHash,
+        BitcoinTx.UTXO calldata walletMainUtxo,
+        bytes20[] calldata targetWallets
+    ) external {
+        Wallet storage wallet = self.registeredWallets[walletPubKeyHash];
+
+        require(
+            wallet.state == WalletState.MovingFunds,
+            "Source wallet must be in MovingFunds state"
+        );
+
+        require(
+            wallet.pendingRedemptionsValue == 0,
+            "Source wallet must handle all pending redemptions first"
+        );
+
+        require(
+            wallet.movingFundsTargetWalletsCommitmentHash == bytes32(0),
+            "Target wallets commitment already submitted"
+        );
+
+        // TODO: Assert the sender is a source wallet operator.
+
+        uint64 walletBtcBalance = getWalletBtcBalance(
+            self,
+            walletPubKeyHash,
+            walletMainUtxo
+        );
+        uint256 expectedTargetWalletsCount = Math.min(
+            self.liveWalletsCount,
+            Math.ceilDiv(walletBtcBalance, self.maxBtcTransfer)
+        );
```

Comment:
Could we add a description how we perform calculations of expected target wallets count like that?

Reply:
This is straight from RFC-1. What exactly do you propose to describe here?